### ISSUE TITLE
[router] [web] - Local dev server returns 404 status code (but correctly renders route) when using a _layout in a dynamic route

### ISSUE THREAD

**Issue by giofilo** [expo/expo#29651]
### Minimal reproducible example

https://github.com/giofilo/expo-dynamic-route-example

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

metro

### Summary

Hi,

as the title says I get 404 error if I launch `npm run web` in the linked repository and try to open one of the two routes:
- http://localhost:8081/[dynamicRoute]/
- http://localhost:8081/[dynamicRoute]/two

where `[dynamicRoute]` could be replaced by everything you want.
The screens are rendered correctly but the status code of the request is 404.
I think the problem is related to the presence of the `[dynamicRoute]/_layout.tsx` file. I will need nested layouts for my use-case.

Thank you for your help!

### Environment

```
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
      Yarn: 1.22.21 - ~/.nvm/versions/node/v20.10.0/bin/yarn
      npm: 10.2.5 - ~/.nvm/versions/node/v20.10.0/bin/npm
      Watchman: 2024.01.22.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2023.1 AI-231.9392.1.2311.11076708
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.12 => 51.0.12 
      expo-router: ~3.5.16 => 3.5.16 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.2 => 0.74.2 
      react-native-web: ~0.19.10 => 0.19.12 
    npmGlobalPackages:
      eas-cli: 7.5.0
    Expo Workflow: managed
```

**Comment by kitten** [comment]
This seems to have been resolved as well by #29948.
It likely has been part of a regression that's also described in #29883. Sorry about that! The fix mentioned has been merged and the next patch release of `expo-router` will fix this.

**Comment by giofilo** [comment]
Thank you!
